language_clean_patch.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+from unidiff import PatchSet
+from unidiff.constants import LINE_TYPE_EMPTY
+
+class PatchCleaner:
+    """
+    Cleans a given unified diff such that only lines matching the language parameter are included,
+    as well as any preceding lines that may be giving way to them (e.g. by adding a trailing comma).
+    """
+
+    def __init__(self, filename, language):
+        self.patch = PatchSet.from_filename(filename)
+        self.language = language
+        self.clean_patch()
+
+    def __str__(self):
+        return str(self.patch)
+
+    def clean_patch(self):
+        """ Cleans the patch, removing irrelevant files. """
+
+        files_to_be_removed = []
+        for i, file in enumerate(self.patch):
+            self.clean_file(file)
+
+            if not file.added and not file.removed:
+                files_to_be_removed.append(i)
+
+        # Any files to be removed?
+        # (In reverse order, as we'll be reindexing.)
+        if len(files_to_be_removed):
+            for i in reversed(files_to_be_removed):
+                del self.patch[i]
+
+    def clean_file(self, file):
+        """ Cleans one particular file in the patch set, removing empty hunks. """
+
+        hunks_to_be_removed = []
+        for j, hunk in enumerate(file):
+            self.clean_hunk(hunk)
+
+            # Is this hunk still modifying anything?
+            # If not, we'll drop it after iterating everything.
+            if not hunk.added and not hunk.removed:
+                hunks_to_be_removed.append(j)
+
+        # Any hunks to be removed?
+        # (In reverse order, as we'll be reindexing.)
+        if len(hunks_to_be_removed):
+            for j in reversed(hunks_to_be_removed):
+                del file[j]
+
+    def clean_hunk(self, hunk):
+        """ Cleans one particular hunk in the patch set, removing irrelevant lines. """
+
+        prev_line = None
+        for k, line in enumerate(hunk):
+            # Is this line modifying anything?
+            if line.is_added or line.is_removed:
+                # We're definitely keeping lines that have something to do
+                # with the target language.
+                if self.language in line.value:
+                    continue
```

Comment:
Doesn't the dump script dumps missing strings on the target language as equal to the reference language?

This will add them back (but I'm not sure how to circumvent this for now)

Reply:
Interesting thought, but due to how the load script works, they wouldn't show up as an addition in the diff right? In case the load script decides to remove a line, this won't change the line status from removal to addition.

Reply:
Ok, if you say so. Just good to keep in mind in case it happens